### PR TITLE
crypto: Add a function to run OPAL PSID reset

### PR DESCRIPTION
**Pull request by vojtechtrefny**

### CODE REVIEW

File: src/plugins/crypto.c [hunk 1]
```diff
@@ -3827,3 +3827,92 @@ gboolean bd_crypto_opal_format (const gchar *device, const gchar *cipher, guint6
...
+        g_propagate_error (error, l_error);
+        return FALSE;
+    }
+
+    if (!bd_crypto_opal_is_supported (device, &l_error)) {
+        g_prefix_error (&l_error, "OPAL doesn't seem to be supported on %s: ", device);
+        bd_utils_report_finished (progress_id, l_error->message);
+        g_propagate_error (error, l_error);
+        return FALSE;
+    }
```
Comment (tbzatek) [inline src/plugins/crypto.c:3878]: Leaking `cd` - move this block before `crypt_init()`?

File: src/plugins/crypto.c [hunk 2]
```diff
@@ -3827,3 +3827,92 @@ gboolean bd_crypto_opal_format (const gchar *device, const gchar *cipher, guint6
...
+
+    if (context->type == BD_CRYPTO_KEYSLOT_CONTEXT_TYPE_PASSPHRASE) {
+        key_buf = (char *) context->u.passphrase.pass_data;
+        buf_len = context->u.passphrase.data_len;
+    } else if (context->type == BD_CRYPTO_KEYSLOT_CONTEXT_TYPE_KEYFILE) {
+        ret = crypt_keyfile_device_read (cd, context->u.keyfile.keyfile, &key_buf, &buf_len,
+                                         context->u.keyfile.keyfile_offset, context->u.keyfile.key_size, 0);
+        if (ret != 0) {
+            g_set_error (&l_error, BD_CRYPTO_ERROR, BD_CRYPTO_ERROR_KEYFILE_FAILED,
+                         "Failed to read key from file '%s: %s", context->u.keyfile.keyfile,
```
Comment (tbzatek) [inline src/plugins/crypto.c:3888]: Stray apostrophe?

Comment (vojtechtrefny) [reply src/plugins/crypto.c]: More than one :-)

**Review by tbzatek — APPROVED** [review]
Good to go now!

**Pull request merged**